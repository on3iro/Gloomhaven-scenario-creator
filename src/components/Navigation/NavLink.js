/**
  * Renders a single NavLink item containing an li and a preformatted react-router Link
  *
  * @namespace Navigation.NavLink
  * @memberOf Navigation
  */

import React, { PropTypes } from 'react';
import styled from 'styled-components';

import StyledLink from 'components/StyledLink';


export const Li = styled.li`
  float: left;
  display: inline;

  &:hover {
    background-color: #666;
  }
`;

export const RestyledLink = styled(StyledLink)`
  color: #fff;
`;

export const NavLink = props => {
  return (
    <Li>
      <RestyledLink to={props.to}>{props.children}</RestyledLink>
    </Li>
  );
};

NavLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

export default NavLink;
