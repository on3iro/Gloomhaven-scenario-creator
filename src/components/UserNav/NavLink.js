/**
  * Renders a single NavLink item containing an li and a preformatted react-router Link
  *
  * @namespace Navigation.NavLink
  * @memberOf Navigation
  */

import React, { PropTypes } from 'react';
import styled from 'styled-components';

import StyledLink from 'components/StyledLink';


export const Li = styled.li`
  float: left;
  display: inline;

  &:hover {
    background-color: #666;
  }
`;

const NavLink = props => {
  return (
    <Li>
      <StyledLink to={props.to}>{props.children}</StyledLink>
    </Li>
  );
};

NavLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.string.isRequired,
};

export default NavLink;
