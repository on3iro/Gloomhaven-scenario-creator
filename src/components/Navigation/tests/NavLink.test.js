import React from 'react';
import { shallow } from 'enzyme';

import NavLink, { Li, RestyledLink } from '../NavLink';



describe('<NavLink />', () => {
  it('Should render react-router li', () => {
    const renderedWrapper = shallow(<NavLink to="/">Home</NavLink>);
    expect(renderedWrapper.find(Li).length).toMatchSnapshot();
  });

  it('should render RestyledLink', () => {
    const renderedWrapper = shallow(<NavLink to="/">Home</NavLink>);
    expect(renderedWrapper.find(RestyledLink).length).toMatchSnapshot();
  });
});
