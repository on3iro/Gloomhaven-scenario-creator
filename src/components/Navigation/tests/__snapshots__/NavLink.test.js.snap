exports[`<NavLink /> Should render react-router li 1`] = `1`;

exports[`<NavLink /> should render RestyledLink 1`] = `1`;
